### PR TITLE
added new route for getDeviceTime

### PR DESCRIPTION
**Pull request by SrinivasanTarget**
@Jonahss opened this to enable https://github.com/appium/appium-ios-driver/pull/59

### CODE REVIEW

**Comment by SrinivasanTarget** [comment]
Error doesn't seems to be related to my code changes

**Comment by imurchie** [comment]
The error in the build is because the tests check if the protocol has changed, to maintain compatibility. You need to update https://github.com/appium/node-mobile-json-wire-protocol/blob/0b7d9b6587ac750df53daaf87768cdf31bfaf7c9/test/routes-specs.js#L41 with the new hash (`d8ffd6a1` in this case).

**Comment by imurchie** [comment]
Looks good to me. Can you squash into one commit.

We will then need to make issues to update all the clients with a method for this endpoint.

**Comment by SrinivasanTarget** [comment]
@imurchie Its squashed now

**Comment by SrinivasanTarget** [comment]
@imurchie can this be merged?

**Comment by imurchie** [comment]
I'd like another reivew. @Jonahss ?

**Comment by Jonahss** [comment]
Since this affects the api I'd like to ask @jlipps to approve it. But it looks good to me :)

File: lib/routes.js [hunk 1]
```diff
@@ -291,6 +291,9 @@ const METHOD_MAP = {
   '/wd/hub/session/:sessionId/appium/device/shake': {
     POST: {command: 'mobileShake'}
   },
+  '/wd/hub/session/:sessionId/appium/device/system_time': {
+    GET: {command: 'getDeviceDateAndTime'}
```
Comment (jlipps) [inline lib/routes.js]: do you think that 'time' implies 'date' here? if so I'd prefer the less verbose 'getDeviceTime'

**Comment by jlipps** [comment]
one comment otherwise LGTM

**Comment by SrinivasanTarget** [comment]
@imurchie can you help me on this?

**Comment by imurchie** [comment]
Sure. What do you need help with?

The error in CI is the same as the earlier one. The hash needs to be updated to `16d9dfdb` in https://github.com/appium/node-mobile-json-wire-protocol/blob/0b7d9b6587ac750df53daaf87768cdf31bfaf7c9/test/routes-specs.js#L41

**Comment by SrinivasanTarget** [comment]
Thanks @imurchie but what does this hash mean?

**Comment by imurchie** [comment]
The hash is essentially a number that represents the state of the protocol. If the protocol changes, then the hash will change. The test just makes sure that we intended to change the protocol.

**Comment by SrinivasanTarget** [comment]
Yup Got it now (y)

**Comment by SrinivasanTarget** [comment]
Squashed again

**Comment by imurchie** [comment]
Thanks for this! We'll get the device implementations in as soon as things settle down!

**Pull request merged**